server/tilesource/base.py

```diff
@@ -15,14 +15,28 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
-###############################################################################
+#############################################################################
 
+import math
+
+from girder import logger
 from girder.models.model_base import ValidationException
 from girder.utility import assetstore_utilities
 from girder.utility.model_importer import ModelImporter
+from six import BytesIO
 
 from ..models.base import TileGeneralException
 
+# Not having PIL disables thumbnail creation, but isn't fatal
```

Comment:
This is fine for now, but we should start thinking about how to handle graceful degradation in a more organized way.